src/plugins/template/src/infer_request.cpp

```diff
@@ -0,0 +1,253 @@
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name + "_Postprocess"),
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name +
+                              "_StartPipline"),
+        openvino::itt::handle("Template" + std::to_string(compiled_model->_cfg.deviceId) + "_" + name + "_WaitPipline"),
+    };
+
+    m_executable = compiled_model->get_template_plugin()->_backend->compile(compiled_model->m_model);
+
+    // Allocate plugin backend specific memory handles
+    m_plugin_input_tensors.resize(get_inputs().size());
+    m_plugin_output_tensors.resize(get_outputs().size());
+    m_input_tensors.resize(get_inputs().size());
+    m_output_tensors.resize(get_outputs().size());
+
+    AllocateImpl(m_input_tensors, get_inputs());
+    AllocateImpl(m_output_tensors, get_outputs());
+}
+// ! [infer_request:ctor]
+
+std::vector<ov::VariableState> TemplatePlugin::InferRequest::query_state() const {
+    OPENVINO_NOT_IMPLEMENTED;
+}
+
+std::shared_ptr<const TemplatePlugin::CompiledModel> TemplatePlugin::InferRequest::get_template_model() const {
+    auto& compiled_model = get_compiled_model();
+    auto template_model = std::dynamic_pointer_cast<const TemplatePlugin::CompiledModel>(compiled_model);
+    OPENVINO_ASSERT(template_model);
+    return template_model;
+}
+
+// ! [infer_request:dtor]
+TemplatePlugin::InferRequest::~InferRequest() {
+    auto compiled_model = std::const_pointer_cast<TemplatePlugin::CompiledModel>(get_template_model());
+    compiled_model->_requestId--;
+}
+// ! [infer_request:dtor]
+
+// ! [infer_request:infer_impl]
+void TemplatePlugin::InferRequest::infer() {
+    // TODO: fill with actual list of pipeline stages, which are executed synchronously for sync infer requests
+    infer_preprocess();
+    start_pipeline();
+    wait_pipeline();  // does nothing in current implementation
+    infer_postprocess();
+}
+// ! [infer_request:infer_impl]
+
+// ! [infer_request:infer_preprocess]
+void TemplatePlugin::InferRequest::infer_preprocess() {
+    OV_ITT_SCOPED_TASK(itt::domains::TemplatePlugin, m_profiling_task[Preprocess]);
+    auto start = Time::now();
+    convert_batched_tensors();
+
+    OPENVINO_ASSERT(m_input_tensors.size() == m_plugin_input_tensors.size());
+    for (size_t i = 0; i < m_input_tensors.size(); i++) {
+        // No ROI extraction is needed
+        m_plugin_input_tensors[i] = get_template_model()->get_template_plugin()->_backend->create_tensor(
+            m_input_tensors.at(i).get_element_type(),
+            m_input_tensors.at(i).get_shape(),
+            m_input_tensors.at(i).data());
+    }
+    OPENVINO_ASSERT(m_output_tensors.size() == m_plugin_output_tensors.size());
+    for (size_t i = 0; i < m_output_tensors.size(); i++) {
+        m_plugin_output_tensors[i] = get_template_model()->get_template_plugin()->_backend->create_tensor(
```

Comment:
maybe infer request itself can store refereence to backend? to shorten this line